### DISCUSSION TITLE
[Feature] Play downloaded full-res video instead of remote low-res preview

### DISCUSSION THREAD

**Discussion by almogtzabari** [immich-app/immich#9219] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

In Google Photos, if you open a video that is not stored locally on your device, it will play a low-resolution preview version. However, once you download it to your device (and close and re-open the video), it will play the original full-resolution version.

In contrast, when you download a video via Immich, it downloads it in full resolution. However, if you try to play it using the Immich app, it won’t play the full-resolution version; instead, it plays the same preview, low-res version that was played prior to downloading. This means that if you’d like to play the original full-res version, you will have to play it via the built-in Photos app (and not via Immich).

Therefore, this feature request is to let immich play the original full-resolution video version if it is present locally on the mobile device, and only if it is not present, play the low-res preview version.

Note: It was mentioned on discord that it might be hard to figure out whether the original video could be played on the device, or not. I'm not sure how would you feel about it, but personally I'd also be fine with a toggle-able button like "Play original version for downloaded videos" or "Prefer local videos over remote ones when playing videos", or something like that. This way, people will be able to disable this feature if they're having trouble playing downloaded versions.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by AngelaDMerkel** [comment]
> "Play original version for downloaded videos" or "Prefer local videos over remote ones when playing videos", or something like that. This way, people will be able to disable this feature if they're having trouble playing downloaded versions.

I think this is the better approach here. I have some very high resolution video and sometimes in formats which require CPU decoding as in 12bit or higher video. Having that play from a local copy once downloaded isn't ideal

**Comment by bo0tzz** [comment]
I'm pretty sure the app behaves as requested these days (#21119)
